node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -478,82 +483,268 @@ mod tests {
         }
     }
 
-    fn unwrap_single<T: Debug>(vec: Vec<T>) -> T {
-        let mut iter = vec.into_iter();
+    struct TestState {
+        rng: TestRng,
+        state: State<TestContext>,
+        instance_id: u64,
+        fd: FinalityDetector<TestContext>,
+        validators: HashMap<ValidatorIndex, ActiveValidator<TestContext>>,
+        timers: BTreeSet<(Timestamp, ValidatorIndex)>,
+    }
+
+    impl TestState {
+        fn new(
+            state: State<TestContext>,
+            rng: TestRng,
+            start_time: Timestamp,
+            instance_id: u64,
+            fd: FinalityDetector<TestContext>,
+            validators: Vec<ValidatorIndex>,
+        ) -> Self {
+            let mut timers = BTreeSet::new();
+            let earliest_round_start = state::round_id(start_time, state.params().init_round_exp())
+                + (1 << state.params().init_round_exp()).into();
```

Comment:
…except if `start_time` is divisible by the initial round length?
```rust
state::round_id(
    start_time + ((1 << state.params().init_round_exp()) - 1).into(),
    state.params().init_round_exp()
);
```

Reply:
Yes, I just changed it this morning. Will push it shortly.

Reply:
Fixed in [`3e59e5b` (#602)](https://github.com/CasperLabs/casper-node/pull/602/commits/3e59e5be187011e0da8f7801704d291ba64967fc).